packages/web-app-files/src/store/actions.ts

```diff
@@ -236,49 +236,21 @@ export default {
     context,
     { client, path, shareWith, shareType, permissions, role, expirationDate, storageId }
   ) {
-    if (shareType === ShareTypes.group.value) {
-      client.shares
-        .shareFileWithGroup(path, shareWith, {
-          permissions,
-          role: role.name,
-          expirationDate,
-          spaceRef: storageId
-        })
-        .then((share) => {
-          context.commit(
-            'CURRENT_FILE_OUTGOING_SHARES_UPSERT',
-            buildCollaboratorShare(
-              share.shareInfo,
-              context.getters.highlightedFile,
-              allowSharePermissions(context.rootGetters)
-            )
-          )
-          context.dispatch('updateCurrentFileShareTypes')
-          context.dispatch('loadIndicators', { client, currentFolder: path })
-        })
-        .catch((e) => {
-          context.dispatch(
-            'showMessage',
-            {
-              title: $gettext('Error while sharing.'),
-              desc: e,
-              status: 'danger'
-            },
-            { root: true }
-          )
-        })
-      return
+    const isGroupShare = shareType === ShareTypes.group.value
```

Comment:
Could you add unit tests for this? At least checking that the correct (mocked) function of the sdk is called would be good.

Why can you omit the mutation completely?

Reply:
Sure 👍  Which mutation do you mean?

Reply:
A few lines below there was a `CURRENT_FILE_OUTGOING_SHARES_UPSERT` mutation.

Reply:
It's still present (a few lines below), I just removed the code duplication.

Reply:
Whoops. Wanted to say "didn't see it in the diff", but it actually is the last visible line 😆